src/LaravelDataTables/Providers/ServiceProvider.php

```diff
@@ -24,14 +30,52 @@ public function boot()
             $dataTable->setConfig($configuration);
             $request = Request::duplicate();
 
-            if ($request->get('csv') == 'prepare') {
+            if (in_array($request->get('csv'), ['prepare', 'email'])) {
                 $request->merge([
                     'start' => null,
                     'length' => null,
                 ]);
 
-                $csv = $dataTable->setRequest($request)
-                    ->makeResponseCsv();
+                $dataTable->setRequest($request);
+
+                if($request->get('csv') == 'email') {
+                    $user = Auth::user();
+                    $params = [
+                        'userId' => $user->getId(),
+                        'configuration' => $configuration,
+                        'filters' => $request->all(),
+                        'title' => $dataTable->getConfig()->getTitle() ??
+                            preg_replace('|^(.*?)DataTable$|s', '$1', basename(get_class($configuration))),
+                    ];
+
+                    dispatch(function () use ($params) {
```

Comment:
I'm assuming we did this as a callable rather than a Job class for a reason and a Job class had problems?

Reply:
Both DataTable and Config classes have serialization problems caused by having closure properties, as well as some other classes that also fail to serialize. There was a tiny opportunity to use jobs if we pass the config class name + filters, so the datatable could be freshly instantiated and called in the job's handle(), but some datatables also have different constructor parameters, that would complicate this further. 
For example, the datatable in **Http/Controllers/Billing/PricingReportController.php**:
`$dataTable = new PricingDataTable(Auth::user()->getCustomer());`
`return response()->dataTable($dataTable);`

Auth::user()->getCustomer() should be passed to the dataTable macro (that means modifying it in all controller actions), or if somehow use the reflection class to get a list of parameters and values an object constructor is called with, that I'm not sure is possible.

Reply:
That's fine, I figured you'd tried and settled on this solution, just wanted to understand why. With laravel searlising the job into a closure like this we avoid all the problems?